pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -49,131 +52,52 @@ import (
 // These are intentionally pre-computed based on events such that lookups are efficient.
 type AmbientIndex struct {
 	mu sync.RWMutex
-	// byService indexes by Service (virtual) *IP address*. A given Service may have multiple IPs, thus
-	// multiple entries in the map. A given IP can have many workloads associated.
-	byService map[string][]*model.WorkloadInfo
-	// byPod indexes by Pod IP address.
-	byPod map[string]*model.WorkloadInfo
-
-	// Map of ServiceAccount -> IP
-	// TODO: currently, this is derived from pods. To be agnostic to the implementation,
-	// we should actually be looking at Gateway.status.addresses.
-	// This may be an external address (possibly even a DNS name we need to resolve), an arbitrary IP,
-	// or a reference to a service.
-	// If its a reference to a Service then we can find the underlying pods in that service, as an optimization.
-	waypoints map[model.WaypointScope]sets.String
-
-	// serviceVipIndex maintains an index of VIP -> Service
-	serviceVipIndex *kclient.Index[*v1.Service, string]
+
+	// podIPIndex maintains an index of IP -> Pod
+	podIPIndex *kclient.Index[*v1.Pod, string]
```

Comment:
i think it's a little confusing that the types to `kclient.Index` are reversed in order from a typical map. even though this functions as IP -> Pod, the types here are Pod -> IP string

Reply:
https://github.com/istio/istio/pull/44667 :+1: thanks!